chirp/drivers/uvk5.py

```diff
@@ -0,0 +1,1339 @@
+                "Power on display mode",
+                RadioSettingValueList(
+                    WELCOME_LIST,
+                    WELCOME_LIST[tmpdispmode]))
+        basic.append(rs)
+
+        # Keypad Tone
+        tmpkeypadtone = _mem.keypad_tone
+        if tmpkeypadtone >= len(KEYPADTONE_LIST):
+            tmpkeypadtone = 0
+        rs = RadioSetting("keypad_tone", "Keypad tone", RadioSettingValueList(
+            KEYPADTONE_LIST, KEYPADTONE_LIST[tmpkeypadtone]))
+        basic.append(rs)
+
+        # Language
+        tmplanguage = _mem.language
+        if tmplanguage >= len(LANGUAGE_LIST):
+            tmplanguage = 0
+        rs = RadioSetting("language", "Language", RadioSettingValueList(
+            LANGUAGE_LIST, LANGUAGE_LIST[tmplanguage]))
+        basic.append(rs)
+
+        # Alarm mode
+        tmpalarmmode = _mem.alarm_mode
+        if tmpalarmmode >= len(ALARMMODE_LIST):
+            tmpalarmmode = 0
+        rs = RadioSetting("alarm_mode", "Alarm mode", RadioSettingValueList(
+            ALARMMODE_LIST, ALARMMODE_LIST[tmpalarmmode]))
+        basic.append(rs)
+
+        # Reminding of end of talk
+        tmpalarmmode = _mem.reminding_of_end_talk
+        if tmpalarmmode >= len(REMENDOFTALK_LIST):
+            tmpalarmmode = 0
+        rs = RadioSetting(
+                "reminding_of_end_talk",
+                "Reminding of end of talk",
+                RadioSettingValueList(
+                    REMENDOFTALK_LIST,
+                    REMENDOFTALK_LIST[tmpalarmmode]))
+        basic.append(rs)
+
+        # Repeater tail tone elimination
+        tmprte = _mem.repeater_tail_elimination
+        if tmprte >= len(RTE_LIST):
+            tmprte = 0
+        rs = RadioSetting(
+                "repeater_tail_elimination",
+                "Repeater tail tone elimination",
+                RadioSettingValueList(RTE_LIST, RTE_LIST[tmprte]))
+        basic.append(rs)
+
+        # Logo string 1
+        logo1 = str(_mem.logo_line1).strip("\x20\x00\xff")  # +"\x20"*12
+        logo1 = logo1[0:12]
+        rs = RadioSetting("logo1", "Logo string 1 (12 characters)",
+                          RadioSettingValueString(0, 12, logo1))
+        basic.append(rs)
+
+        # Logo string 2
+        logo2 = str(_mem.logo_line2).strip("\x20\x00\xff")  # +"\x20"*12
+        logo2 = logo2[0:12]
+        rs = RadioSetting("logo2", "Logo string 2 (12 characters)",
+                          RadioSettingValueString(0, 12, logo2))
+        basic.append(rs)
+
+        # unlock settings
+
+        # F-LOCK
+        tmpflock = _mem.int_flock
+        if tmpflock >= len(FLOCK_LIST):
+            tmpflock = 0
+        rs = RadioSetting(
+            "flock", "F-LOCK",
+            RadioSettingValueList(FLOCK_LIST, FLOCK_LIST[tmpflock]))
+        unlock.append(rs)
+
+        # 350TX
+        rs = RadioSetting("350tx", "350TX", RadioSettingValueBoolean(
+            bool(_mem.int_350tx > 0)))
+        unlock.append(rs)
+
+        # unknown1
```

Comment:
Please don't expose unknown random bits to the users. If we don't know what they do, we shouldn't be letting the user set them to arbitrary values.

Reply:
We know what they do. The names correspond to the names that the radio shows in it's menus when a special key combination is pressed. I've written better descriptions of the parameters.

Removed the one unknown setting. I think it's not used in the versions of the firmware i've looked at